### PR TITLE
Add a package for posting ModifyAckDeadline messages to GCP periodically

### LINKED ISSUE #777: PubSub messages being redelivered while a worker is actively processing the message.
GCP PubSub has a 600 second (10m) window for messages to be Acked.

If the message is not acked in this period it will be scheduled for redelivery.

The workers in Package Analysis can sometimes take over 10m to run, which causes multiple workers to be scheduled to process the same message - effectively starving the cluster of resources over time.

To avoid this the window needs to be greater than 10m, but GCP does not allow values higher than this.

A possible work around is to issue `ModifyAckDeadline` messages to GCP to extend the message's deadline while it is still working on it. Ideally [`time.Ticker`](https://pkg.go.dev/time#Ticker) can be used to deliver updates periodically until the analysis is complete.

### PR DESCRIPTION
**Pull request by calebbrown**
Fixes #777.

### CODE REVIEW

File: cmd/worker/main.go [hunk 1]
```diff
@@ -227,10 +229,24 @@ func messageLoop(ctx context.Context, subURL, packagesBucket, notificationTopicU
 			// All subsequent receive calls will return the same error, so we bail out.
 			return fmt.Errorf("error receiving message: %w", err)
 		}
+		me := extender.Start(ctx, msg, func() {
+			log.Info("Message Ack deadline extended", "message_id", msg.LoggableID, "message_meta", msg.Metadata)
+		})
 
 		if err := handleMessage(ctx, msg, pkgsBkt, resultsBuckets, imageSpec, notificationTopic); err != nil {
 			log.Error("Failed to process message",
 				"error", err)
```
Comment (maxfisher-g) [inline cmd/worker/main.go:238]: nit: remove the linebreak in this function call to be consistent with the other log calls

File: cmd/worker/pubsubextender/extender.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	msg      *pubsub.Message
+	done     chan bool
+	exited   chan error
+	callback func()
+	running  bool
+}
+
+func (e *Extender) Start(ctx context.Context, msg *pubsub.Message, callback func()) *MessageExtender {
+	freq := e.Deadline - e.GracePeriod
+	if freq < 0 {
```
Comment (maxfisher-g) [inline cmd/worker/pubsubextender/extender.go:80]: not sure if this is the most intuitive behaviour, perhaps it should just return an error?

**Review by maxfisher-g — COMMENTED** [review]
Looks mostly good, just a few initial thoughts

File: cmd/worker/main.go (hunk 1 above)
Comment (calebbrown) [reply cmd/worker/main.go]: This is a callback method in the call to `Start()` I'm not sure Go's formatter likes removing those linebreaks.

Comment (maxfisher-g) [reply cmd/worker/main.go]: Oh iThe commit introducing the change is [this one]( https://github.com/composer/getcomposer.org/commit/70527179915d55b3811bebaec55926afd331091b)

Comment (maxfisher-g) [reply cmd/worker/main.go]: Oh, I didn't see the callback part, no worries then

**Review by maxfisher-g — APPROVED** [review]

**Pull request merged**